### PR TITLE
Added go client generator

### PR DESCRIPTION
**Pull request by airon-applyinnovations**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,7 +8,7 @@ hospitality-api-docs
 .envrc
 
 dist
-src
+src/typescript-fetch
 
 *.log
```
Comment (alexander-applyinnovations) [inline .gitignore:13]:
```suggestion
!src/go/oauth/authenticationMiddleware.go
```

File: templates/go/api.mustache [hunk 2]
```diff
@@ -244,9 +236,6 @@ func (a *{{{classname}}}Service) {{nickname}}Execute(r {{#structPrefix}}{{&class
 		localVarHeaderParams["Accept"] = localVarHTTPHeaderAccept
 	}
 {{#headerParams}}
-	{{#required}}
-	parameterAddToHeaderOrQuery(localVarHeaderParams, "{{{baseName}}}", r.{{paramName}}, "{{collectionFormat}}")
-	{{/required}}
 	{{^required}}
```
Comment (alexander-applyinnovations) [inline templates/go/api.mustache]: edit this to add header or query for both required and not required

File: src/typescript-fetch/authenticationMiddleware.ts [hunk 3]
```diff
@@ -38,22 +40,27 @@ export class OhipCredentialsProvider {
...
-      this.access_token = access_token;
-      this.expiry = expiry;
-    }
   }
 
-  async loadCredentials() {
-    // do nothing
-    // optionally load from persistent storage
-  }
+  async setAccessToken(access_token: string) {
```
Comment (airon-applyinnovations) [inline src/typescript-fetch/authenticationMiddleware.ts:45]: added ability to set access token, will only be set if it's not expired or is expiration is later than the current token.

File: src/typescript-fetch/authenticationMiddleware.ts [hunk 4]
```diff
@@ -19,13 +24,10 @@ export class OhipCredentialsProvider {
   appKey: string;
   credentials: OhipCredential[];
   access_token?: string;
-  expiry?: number;
```
Comment (airon-applyinnovations) [inline src/typescript-fetch/authenticationMiddleware.ts:22]: expiry is now derived from JWT. it's epoch time which can be more accurate than doing `new Date() + expiresIn`. this also simplify things as we don't have to keep updating both access_token and expiry.

File: package.json [hunk 5]
```diff
@@ -6,7 +6,8 @@
   "scripts": {
     "reset-docs": "cd hospitality-api-docs && git reset --hard && cd ..",
     "clean": "find src -type f ! -name '*authenticationMiddleware.*' -exec rm {} + && find src -type d -empty -delete && rm -rf dist",
-    "build": "openapi-generator-cli generate > build.log",
+    "add-auth": "cp src/go/authentication_middleware.go dist/go/oauth/",
```
Comment (alexander-applyinnovations) [inline package.json]: Just copy all the contents of the src folder into the dist folder and move your file into a subdirectory

**Review by alexander-applyinnovations — CHANGES_REQUESTED** [review]

**Review by alexander-applyinnovations — APPROVED** [review]

**Pull request merged**